### DISCUSSION TITLE
is it adaptive config for vmess AuthenticatedLength?

### DISCUSSION THREAD

**Discussion by rurirei** [v2fly/v2ray-core#2319] [Q&A] [answered]
>> experiments: string
启用的 VMess 协议实验性功能。（此处的功能为不稳定功能， 可能随时被移除）多个启用的实验之间可以用 | 字符分割，如 "AuthenticatedLength|NoTerminationSignal" 。
"AuthenticatedLength" 启用认证的数据包长度实验。此实验需要同时在客户端与服务器端同时开启，并运行相同版本的程序。 (v4.41.0+)
"NoTerminationSignal" 启用不发送断开连接标致实验。此实验可能会影响被代理的连接的稳定性。 (v4.41.0+)

>> This experiment needs to be opened on both the client and the server at the same time, and run the same version of the program.

is it adaptive for server-side to receive request with option vmess AuthenticatedLength, though the doc says it's non-adaptive as "This experiment needs to be opened on both the client and the server"? @xiaokangwang 

https://github.com/v2fly/v2ray-core/blob/a3a9e46069f2c769d1e496bc3cbedd42f36b21af/proxy/vmess/outbound/outbound.go#L123-L125

https://github.com/v2fly/v2ray-core/blob/a3a9e46069f2c769d1e496bc3cbedd42f36b21af/proxy/vmess/encoding/server.go#L358

https://github.com/v2fly/v2ray-core/blob/a3a9e46069f2c769d1e496bc3cbedd42f36b21af/testing/scenarios/vmess_test.go#L1414

**Comment by xiaokangwang** [comment, marked as answer]
When this option is set, RequestOptionAuthenticatedLength will signal that this connection should use AuthenticatedLength extension. https://github.com/v2fly/v2ray-core/blob/master/common/protocol/headers.go#L44

If the server supports the same version of AuthenticatedLength extension, then connection  will work. No server side setting is necesary.
